tt_metal/llrt/tt_cluster.cpp

```diff
@@ -607,6 +634,100 @@ std::tuple<chip_id_t, CoreCoord> Cluster::get_connected_ethernet_core(std::tuple
         std::get<0>(connected_eth_core), soc_desc.chan_to_logical_eth_core_map.at(std::get<1>(connected_eth_core)));
 }
 
+CoreCoord Cluster::ethernet_core_from_logical_core(chip_id_t chip_id, const CoreCoord &logical_core) const {
+    const metal_SocDescriptor &soc_desc = get_soc_desc(chip_id);
+    return soc_desc.get_physical_ethernet_core_from_logical(logical_core);
+}
+
+void Cluster::set_routing_config_for_ethernet_cores() {
+    const char *TT_METAL_SLOW_DISPATCH_MODE = std::getenv("TT_METAL_SLOW_DISPATCH_MODE");
```

Comment:
beyond the scope of this change, but we should move the env var into the settings and pull that rather than looking up a string...